src/number_theory/padics/padic_integers.lean

```diff
@@ -579,6 +578,27 @@ lemma ideal_eq_span_pow_p {s : ideal ℤ_[p]} (hs : s ≠ ⊥) :
   ∃ n : ℕ, s = ideal.span {p ^ n} :=
 discrete_valuation_ring.ideal_eq_span_pow_irreducible hs irreducible_p
 
+open cau_seq
+
+instance : is_adic_complete (maximal_ideal ℤ_[p]) ℤ_[p] :=
+{ prec' := λ x hx,
+  begin
+    simp only [← ideal.one_eq_top, smul_eq_mul, mul_one, smodeq.sub_mem, maximal_ideal_eq_span_p,
+      ideal.span_singleton_pow, ← norm_le_pow_iff_mem_span_pow] at hx ⊢,
+    let x' : cau_seq ℤ_[p] norm := ⟨x, _⟩, swap,
+    { intros ε hε, obtain ⟨m, hm⟩ := exists_pow_neg_lt p hε,
+      refine ⟨m, λ n hn, lt_of_le_of_lt _ hm⟩, rw [← neg_sub, norm_neg], exact hx hn },
+    { refine ⟨x'.lim, λ n, _⟩,
+      have : (0:ℝ) < p ^ (-n : ℤ), { apply _root_.fpow_pos_of_pos, exact_mod_cast hp_prime.1.pos },
```

Comment:
I would rather remove the `open nat` and this `_root_`. In my experience `open nat` is the single most effective name collision generator we have in mathlib. In this file it used once in a `cast_injective` that could be a `nat.cast_injective` on line 160 and once on line 481 using a specialized `nat` lemma that could be replace by its general version as in `have : p ^ n ≤ p ^ k ↔ n ≤ k := (strict_mono_pow hp_prime.1.one_lt).le_iff_le,`

bors d+

Reply:
Thanks, fixed.